pkg/satellite/satellitedb/db.go

```diff
@@ -12,6 +15,9 @@ import (
 // Database contains access to different satellite databases
 type Database struct {
 	db *dbx.DB
+	tx *dbx.Tx
+
+	methods dbx.Methods
```

Comment:
a few words about this implementation.

All methods signatures described in dbx file will be added to dbx.Methods interface (also generated by dbx ).
And dbx.DB and dbx.TX both implement it. 

So all repos can work with abstract methods object. And they dont care if i'am working with regular db connection or in the transaction scope

And only thing that i have to do - initialize db.methods field with needed implementation - DB or TX